app/src/main/java/com/amaze/filemanager/ui/activities/MainActivity.java

```diff
@@ -586,6 +595,39 @@ private void initFabToSave(final ArrayList<Uri> uris) {
     floatingActionButton.setVisibility(View.VISIBLE);
   }
 
+  /*
+   * When saving from browser, we are fetching the Summary and the URL of the page and saving it in
+   * a text file and finally returning the URI so that can be saved.
+   * */
+  public Uri createTempTextFile(String data, String textFileName) {
+    try {
+      String rootPath =
+          Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).getPath();
```

Comment:
What I would suggest is to let user select the save location with the FAB download button on click. Don't hardcode the text file location to `Environment.DIRECTORY_DOWNLOADS`

Reply:
@VishalNehra how to get the location that the user selected ? I did not find any way of doing that ?

Reply:
I got the solution it was part of the code. 
`getCurrentMainFragment().getCurrentPath()`